### ISSUE TITLE
Shortcut to close All Modules

### ISSUE THREAD

**Issue by PDeb108** [darktable-org/darktable#17072]
Is there a shortcut which will close all open modules in any view? I find it a bit of a chore scrolling up a module to double-click the module header to close it. This becomes especially irritating with those modules that have a lot of options say Color balance RGB as an example, where I have to use the mouse wheel scroll at least 3 times before I reach the header, all to access another module in the same group.  See the picture below.  To access Sigmoid, I have to scroll up several times before I can reach the next module

![image](https://github.com/darktable-org/darktable/assets/165793573/a18e7b68-ece9-4f8e-a520-b79a6daf5d99)

**Comment by ralfbrown** [comment]
No, but there is a preference to close all other modules in the panel when you open one.  And using Shift-Click instead of a simple Click on the module header reverses that preference - if set to close, Shift-Click lets you open multiple modules, while if set not to close, Shift-Click will open the clicked module and close all others.

In the preferences dialog, on the lightroom tab, "expand a single utility module at a time", and on the darkroom tab, "expand a single processing module at a time".  You may also want to turn on "scroll ... modules to the top when expanded" in either or both tabs.

**Comment by PDeb108** [comment]
Yes, I have all of them selected as you suggest.  However the issue I am indicating is somewhat different.  When you open say Color Balance RGB, scroll down to masks, do your work, and then want to access the Sigmoid module, which is in the same group, how can you do it with one click.  Now you have to scroll up 3-4 times the scroll wheel to come to the top of the open module and then get to see the Sigmoid module.  This is true for other modules which have lots of options, like "diffuse or Sharpen"

My request is to have a shortcut that will close all the modules, in this case with one click, I can now access the Sigmoid module by closing the Color Bal....

**Comment by dterrahe** [comment]
`processing modules/<focused>` .default element, `show`, and effect, `toggle`, should generally do what you want. Unless the module lost focus somehow, for example because you changed between module groups. But then it is likely that the module header scrolled into view again, so you can use that to collapse it.
![image](https://github.com/darktable-org/darktable/assets/1549490/82fdb772-27aa-4c3c-9886-10b71898d682)

**Comment by PDeb108** [comment]
In my DT version it is blank below shortcut

![image](https://github.com/darktable-org/darktable/assets/165793573/eeacd571-437d-4797-894c-c0bb3eca47ec)

**Comment by ptilopteri** [comment]
* PDeb108 ***@***.***> [06-29-24 12:59]:
> In my DT version it is blank below shortcut
> 
> ![image](https://github.com/darktable-org/darktable/assets/165793573/eeacd571-437d-4797-894c-c0bb3eca47ec)

just guessing, but appears a borked install

you don't give any information about your install or system :(


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by PDeb108** [comment]
Sorry:
DT Version 4.9+41
Device Configuration:
Device name	LAPTOP-TSGJ4F3I
Processor	11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz   2.42 GHz
Installed RAM	8.00 GB (7.80 GB usable)
Device ID	8B27435A-8086-467F-A43E-1CDD681952B3
Product ID	00356-24525-53642-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Windows Configuration
Edition	Windows 11 Home Single Language
Version	23H2
Installed on: 02-10-2022
OS build	22631.3810

**Comment by ptilopteri** [comment]
* PDeb108 ***@***.***> [06-29-24 21:42]:
> Sorry:
> DT Version 4.9+41
> Device Configuration:
> Device name	LAPTOP-TSGJ4F3I
> Processor	11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz   2.42 GHz
> Installed RAM	8.00 GB (7.80 GB usable)
> Device ID	8B27435A-8086-467F-A43E-1CDD681952B3
> Product ID	00356-24525-53642-AAOEM
> System type	64-bit operating system, x64-based processor
> Pen and touch	No pen or touch input is available for this display
> 
> Windows Configuration
> Edition	Windows 11 Home Single Language
> Version	23H2
> Installed on: 02-10-2022
> OS build	22631.3810


you undoubtedly still have an installation problem as you should have
modules visible below the shortcuts -> processing modules
and you do not.  might be where you obtained the dt package or something
else, your build, ....  I have little knowledge of windows.  


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by PDeb108** [comment]
Github Nightly Build.  Currently on version 4.9+41.  What do I need to do to correct it

**Comment by dterrahe** [comment]
You need to create a new shortcut by double clicking on `<focused>` and then pressing the key you want to assign. Obviously the example I showed is not a _default_ shortcut.

**Comment by ptilopteri** [comment]
* PDeb108 ***@***.***> [06-29-24 23:51]:
> Github Nightly Build.  Currently on version 4.9+41.  What do I need to do to correct it


as I said earlier, I am not familiar with windows, but I would uninstall
the current dt and download and install the present nightly build which is
the version you indicate.  you might want to save your current config
directory.  there are instructions on windows install and location of
particular files.  I would suggest you read it.

this is my screen:
   http://wahoo.no-ip.org/~paka/Screenshot_20240630_082304.png
   
appears your "shortcutsrc" file is not complete, at least.

another approach would be to move your configdir to another location so
you can later restore it, and start dt.  dt will then recreate the
configdir and containing files as original install would have provided.

   
-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by PDeb108** [comment]
My view at the shortcut level is the same as yours.  It differs when I go to the fourth level = shortcuts>processing module><focussed> >buttons.  At this level there are no shortcuts visible.  See the picture in my earlier comment

**Comment by ptilopteri** [comment]
* PDeb108 ***@***.***> [06-30-24 08:41]:
> My view at the shortcut level is the same as yours.  It differs when I go to the fourth level = shortcuts>processing module><focussed> >buttons.  At this level there are no shortcuts visible.  See the picture in my earlier comment

follow dterrahe's instructions

-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc
